binding/nodejs/index.js

```diff
@@ -0,0 +1,205 @@
+
+    let leopardHandleAndStatus = null;
+    try {
+      leopardHandleAndStatus = pvLeopard.init(accessKey, modelPath);
+    } catch (err) {
+      pvStatusToException(PV_STATUS_T[err.code], err);
+    }
+
+    const status = leopardHandleAndStatus.status;
+    if (status !== PV_STATUS_T.SUCCESS) {
+      pvStatusToException(status, "Leopard failed to initialize");
+    }
+
+    this._handle = leopardHandleAndStatus.handle;
+    this._pvLeopard = pvLeopard;
+    this._sampleRate = pvLeopard.sample_rate();
+    this._version = pvLeopard.version();
+  }
+
+  /**
+   * @returns the audio sampling rate accepted by the process function
+   * @see {@link process}
+   */
+  get sampleRate() {
+    return this._sampleRate;
+  }
+
+  /**
+   * @returns the version of the Leopard engine
+   */
+  get version() {
+    return this._version;
+  }
+
+  /**
+   * Processes a given audio data and returns its transcription.
+   *
+   * @param {Array} pcm Audio data. The audio needs to have a sample rate equal to `.sampleRate` and be 16-bit linearly-encoded.
+   * This function operates on single-channel audio. If you wish to process data in a different
+   * sample rate or format consider using `.processFile`.
+   * @returns {string} Inferred transcription.
+   */
+  process(pcm) {
+    if (
+      this._handle === 0 ||
+      this._handle === null ||
+      this._handle === undefined
+    ) {
+      throw new PvStateError("Leopard is not initialized");
+    }
+
+    if (pcm === undefined || pcm === null) {
+      throw new PvArgumentError(
+        `PCM array provided to process() is undefined or null`
+      );
+    } else if (pcm.length === 0) {
+      throw new PvArgumentError(`PCM array provided to process() is empty`);
+    }
+
+    let transcriptAndStatus = null;
+    try {
+      transcriptAndStatus = this._pvLeopard.process(
+        this._handle,
+        pcm,
+        pcm.length
+      );
+    } catch (err) {
+      pvStatusToException(PV_STATUS_T[err.code], err);
+    }
+
+    const status = transcriptAndStatus.status;
+    if (status !== PV_STATUS_T.SUCCESS) {
+      pvStatusToException(status, "Leopard failed to process the audio frame");
+    }
+
+    return transcriptAndStatus.transcript;
+  }
+
+  /**
+   * Processes a given audio file and returns its transcription.
+   *
+   * @param {string} audioPath Absolute path to the audio file.
+   * The file needs to have a sample rate equal to or greater than `.sampleRate`.
+   * The supported formats are: `FLAC`, `MP3`, `Ogg`, `Opus`, `Vorbis`, `WAV`, and `WebM`.
+   * @returns {string} Inferred transcription.
+   */
+  processFile(audioPath) {
+    if (
+      this._handle === 0 ||
+      this._handle === null ||
+      this._handle === undefined
+    ) {
+      throw new PvStateError("Leopard is not initialized");
+    }
+
+    if (!fs.existsSync(audioPath)) {
+      throw new PvArgumentError(
+        `Could not find the audio file at '${modelPath}'`
+      );
+    }
+
```

Comment:
should probably add extension check

Reply:
relying on extensions to identify the audio format can get quite messy and will not always work as people may pass audio files without any extension. So i guess it's better to pass the audio path to C and decoder makes the decision whether it can process it or not

Reply:
hmm, I added checks on mine, I think we talked about this before that we added extension checks just cause I think C side didnt provide helpful errors I think, not sure 100% on this though

Reply:
oh i see, in this case, let me add it as well